src/DoctrineDBALAdapter.php

```diff
@@ -78,23 +79,23 @@ public function fileExists(string $path): bool
 
     public function directoryExists(string $path): bool
     {
-        return $this->exists($this->prefixer->prefixPath($path), self::TYPE_DIR);
+        return $this->exists($this->prefixer->prefixDirectoryPath($path), self::TYPE_DIR);
     }
 
     private function exists(string $prefixedPath, string $type): bool
     {
         return (bool) $this->connection->executeQuery(<<<SQL
-SELECT EXISTS (
+SELECT CASE WHEN EXISTS (
```

Comment:
EXISTS already a boolean operator in SQL.

Reply:
Yes, but for mssql that is inncorect syntax. You can check this on (https://sqliteonline.com/)
```sql
SELECT EXISTS (SELECT 1)
```
throws: Token error: 'Incorrect syntax near the keyword 'exists'.' 

When on Sqlite, PostgreSql, MariaDb, Mysql works like a charm.
Case-When syntax should work on every engine.

But on Oracle still missing `from DUAL`. On this could be patched with simple
```php
$this->connection->getDatabasePlatform()->getDummySelectSQL('CASE WHEN EXISTS(...`)
```

Reply:
getDummySelect wont work. E.g.: DB2.
Currently the bundle supports: sqlite, mysql , mariadb and pgsql.
If we need to add more platform, a refactor is needed to handle platform-specific queries i guess. Not just some if-else in the adapter.

Reply:
The package is great, I just miss support for SQLserver.I'm not intrested with others like Oracle or DB2.

I'm not familiar with DB2. Doctrine do `sprintf('SELECT %s FROM sysibm.sysdummy1', $expression);` for this platform.